ipn/ipnlocal/local.go

```diff
@@ -4067,14 +4067,17 @@ func (b *LocalBackend) setDebugLogsByCapabilityLocked(nm *netmap.NetworkMap) {
 	}
 }
 
-func (b *LocalBackend) reloadServeConfigLocked(prefs ipn.PrefsView) {
+func (b *LocalBackend) reloadServeConfigLocked(prefs ipn.PrefsView, changed bool) {
```

Comment:
what is changed?

Reply:
@bradfitz I noticed that `SetServeConfig` saves to disk, then calls `setTCPPortsInterceptedFromNetmapAndPrefsLocked` which reloads the entire config back from disk. Which has two issues: 
1. Feels inefficient, because we have the config in memory, why read it back from disk again? Not sure, but maybe [this comment](https://github.com/tailscale/tailscale/blob/94304819263b0553d7a93d2c061e8f824233456d/ipn/ipnlocal/local.go#L4079) is related here. 
2. More importantly, this would end up removing the in-memory config so the `changed` bool says "don't actually reload the config from disk" but run the rest of the code, like validation checks or w/e it's doing. Definitely happy to hear better suggestions...or to actually be educated on what this code is doing cuz im flying a lil blind :)

Reply:
Do we still need changed? It's still not documented here.